pkg/service/service.go

```diff
@@ -1664,12 +1664,12 @@ func (s *Service) updateBackendsCacheLocked(svc *svcInfo, backends []lb.Backend)
 				backends[i].State = lb.BackendStateActive
 				newBackends = append(newBackends, backends[i])
 				// TODO make backendByHash by value not by ref
-				s.backendByHash[hash] = &backends[i]
+				s.backendByHash[hash] = backends[i]
```

Comment:
The following change is needed:
```suggestion
				s.backendByHash[hash] = backends[i]
				svcBackendsById[backend.ID] = struct{}{}
```

Reply:
seems like we don't have svcBackendsById in the scope, also the main branch or 1.13 didn't have such a statement. Can you elaborate a little bit more?

https://github.com/cilium/cilium/blob/b560c230452573731faae05ddf17dc4231b4ca8b/pkg/service/service.go#L1640-L1641